docs/static/configuration.asciidoc

```diff
@@ -599,6 +599,199 @@ output {
 }
 ----------------------------------
 
+[[environment-variables]]
+=== Using Environment Variables in Configuration
+== Overview
+
+You can set environment variable references into logstash plugins configuration using `${var}` or `$var`. 
+Each reference will be replaced by environment variable value at logstash startup.
+
+* The replacement is case-sensitive.
+* References to undefined variables raise a logstash configuration error.
+* A default value can be given by using the form `${var:default value}`.
+* You can add environment variable references in any plugin option type : string, number, boolean, array or hash.
```

Comment:
Can you also add that environment variables are immutable. If you update the environment variable, you'll have to restart Logstash to pick the updated value.